app/models/permissions/scientist_permission.rb

```diff
@@ -8,11 +8,40 @@ class ScientistPermission < BasePermission
     def initialize(user)
       super
       allow :scans, [:create]
-      # Check if Scientists should be able to upload labware
       allow :upload_labware, [:create]
+      allow :move_locations, [:create] do |record|
+        check_locations(record)
+        # Users cannot move a protected location
+      end
+      allow :empty_locations, [:create]
+      allow :users, [:update] do |record|
```

Comment:
It feels like there is some validation going on here when maybe it should just be a boolean. I think any validation errors should only be added in the validation. What do you think?